pkg/infrastructure/clusterapi/ignition.go

```diff
@@ -38,43 +38,60 @@ const (
-func EditIgnition(in IgnitionInput, platform string, publicIPAddresses, privateIPAddresses []string) ([]byte, []byte, error) {
+func EditIgnition(in IgnitionInput, platform string, publicIPAddresses, privateIPAddresses []string) (*IgnitionOutput, error) {
 	ignData := &igntypes.Config{}
 	err := json.Unmarshal(in.BootstrapIgnData, ignData)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to unmarshal bootstrap ignition: %w", err)
+		return nil, fmt.Errorf("failed to unmarshal bootstrap ignition: %w", err)
 	}
 
 	err = addLoadBalancersToInfra(platform, ignData, publicIPAddresses, privateIPAddresses)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to add load balancers to ignition config: %w", err)
+		return nil, fmt.Errorf("failed to add load balancers to ignition config: %w", err)
 	}
+	logrus.Debugf("Successfully added API and API-Int Load Balancer IPs to infrastructure manifest")
 
 	publicIPsSingleStr := strings.Join(publicIPAddresses, ",")
 	privateIPsSingleStr := strings.Join(privateIPAddresses, ",")
 
 	lbConfig, err := lbconfig.GenerateLBConfigOverride(privateIPsSingleStr, publicIPsSingleStr)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	if err := asset.NewDefaultFileWriter(lbConfig).PersistToFile(command.RootOpts.Dir); err != nil {
-		return nil, nil, fmt.Errorf("failed to save %s to state file: %w", lbConfig.Name(), err)
+		return nil, fmt.Errorf("failed to save %s to state file: %w", lbConfig.Name(), err)
 	}
+	logrus.Debugf("Successfully added API and API-Int Load Balancer IPs to lbConfig configmap")
+
 	err = updateMCSCertKey(in, platform, ignData, privateIPAddresses)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to update MCS Cert and Key in the bootstrap ignition: %w", err)
+		return nil, fmt.Errorf("failed to update MCS Cert and Key in the bootstrap ignition: %w", err)
 	}
+	logrus.Debugf("Successfully updated bootstrap machine-config-server certs")
 
 	editedIgnBytes, err := json.Marshal(ignData)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to convert ignition data to json: %w", err)
+		return nil, fmt.Errorf("failed to convert ignition data to json: %w", err)
 	}
+	logrus.Debugf("Successfully updated bootstrap ignition with updated manifests")
 
-	editedPointerIgn, err := updatePointerIgnition(in, privateIPAddresses, "master")
+	editedMasterIgn, err := updatePointerIgnition(in, privateIPAddresses, "master")
+	if err != nil {
+		return nil, fmt.Errorf("failed to update Master Pointer Ignition: %w", err)
+	}
+	logrus.Debugf("Successfully updated master pointer ignition with API LB IP")
+	editedWorkerIgn, err := updatePointerIgnition(in, privateIPAddresses, "worker")
 	if err != nil {
-		return editedIgnBytes, in.MasterIgnData, fmt.Errorf("failed to update Pointer Ignition")
+		return nil, fmt.Errorf("failed to update Worker Pointer Ignition: %w", err)
+	}
+	logrus.Debugf("Successfully updated worker pointer ignition with API LB IP")
```

Comment:
nit: I don't mind all of the debug logs here but did you mean to keep these or were they for testing?